appinventor/components/src/com/google/appinventor/components/runtime/util/OnBeforeInitializeListener.java

```diff
@@ -0,0 +1,11 @@
+package com.google.appinventor.components.runtime.util;
+
+/**
+ * This interface contains a callback method onBeforeInitialize()
+ * which will be invoked by the Form activity after the app is
+ * initialized, but before the Screen Initialize event is dispatched.
```

Comment:
@ewpatton I think there was a discussion on this a while back (#1748), and this solves some very specific issues (which maybe we can avoid in the first place; I'll have a look at it again)

The reasoning behind this was basically that there may be some internal variables that keep track of something in the component. Now, the thing is, some blocks may rely on these internal variables, and then those internal variables should only be set onInitialize. Since this is the case, the blocks that rely on those variables will access them before the Initialize is called (because Initialize is only called after the "Initialize" block). So if a user happens to use those blocks in the Initialize block, it will lead to undefined behavior.

Reply:
@ewpatton You are completely right; It somehow completely went over my head that there was another Initialize method (I kept confusing it with the other onInitialize method) that gets called from .scm. I will remove this interface, since Initialize seemed to have exactly the same result.